include/nuttx/sched_note.h

```diff
@@ -136,14 +141,24 @@
           sched_note_bprintf_ip(tag, SCHED_NOTE_IP, event, \
                                 fmt, ##__VA_ARGS__)
 #  define sched_note_beginex(tag, str) \
-          sched_note_printf_ip(tag, SCHED_NOTE_IP, "B|%d|%s", _SCHED_GETTID(), str)
+          sched_note_dump_ip(tag, SCHED_NOTE_IP, NOTE_BEGINEX_EVENT, \
+                             str, strlen(str))
 #  define sched_note_endex(tag, str) \
-          sched_note_printf_ip(tag, SCHED_NOTE_IP, "E|%d|%s", _SCHED_GETTID(), str)
+          sched_note_dump_ip(tag, SCHED_NOTE_IP, NOTE_ENDEX_EVENT, \
+                             str, strlen(str))
 #  define sched_note_mark(tag, str) \
-          sched_note_printf_ip(tag, SCHED_NOTE_IP, "I|%d|%s", _SCHED_GETTID(), str)
+          sched_note_dump_ip(tag, SCHED_NOTE_IP, NOTE_MARK_EVENT, \
+                             str, strlen(str))
 #  define sched_note_counter(tag, name, value) \
-          sched_note_printf_ip(tag, SCHED_NOTE_IP, "C|%d|%s|%" PRId32, \
-                               _SCHED_GETTID(), name, value)
+          do \
+            { \
+              struct note_counter_s _counter_ = \
+                { \
+                  name, value \
```

Comment:
name may disappear in noteram_dump_one, let's change to:
```
#define sched_note_counter(tag, name, value) \
             sched_note_bprintf(tag, NOTE_EVENT_COUNTER, "%lx%s", (unsigned long)value, name)
```
and implement %s inside sched_note_vbprintf.